src/main/java/uk/gov/ons/ssdc/caseprocessor/messaging/UpdateSampleSensitiveReceiver.java

```diff
@@ -63,9 +63,11 @@ public void receiveMessage(Message<byte[]> message) {
         caze, "Sensitive data updated", EventType.UPDATE_SAMPLE_SENSITIVE, event, message);
   }
 
-  private void validateOnlySensitiveDataBeingUpdated(Case caze, Entry<String, String> entry) {
-    if (!caze.getSampleSensitive().containsKey(entry.getKey())) {
-      throw new RuntimeException("Key (" + entry.getKey() + ") does not match an existing entry!");
+  private void validateOnlySensitiveDataBeingUpdated(Case caze, String columnName) {
+    //    TODO: Is this right?  Should it not be within the sampleSensitive definition?
+//    Adding new data, if within sample def should be ok, if we have non mandatory fields?
```

Comment:
Agreed, I think this currently appears to work fine because when loading a sample from a CSV, all the sensitive fields get populated, just with empty strings. If we were to create the case by a different channel which didn't include some sensitive fields at all, or update it to remove a field from the sensitive JSON entirely, this would be broken.